src/net/inv.rs

```diff
@@ -0,0 +1,1509 @@
+                        if sn.block_height < highest_block_height {
+                            test_debug!("{:?}: neighbor {:?} has processed only up to burn block {} (we are targeting {})", &self.local_peer, nk, sn.block_height, target_block_height);
+
+                            highest_block_height = sn.block_height;
+                            num_blocks = if highest_block_height >= target_block_height { highest_block_height - target_block_height } else { 0 };
+                        }
+                    },
+                    // this peer is unstable -- its tip consensus hash differs from ours, but we
+                    // agree with its stable consensus hash.  Ask only for blocks no higher than
+                    // its stable consensus hash.
+                    None => match BurnDB::get_block_snapshot_consensus(burndb.conn(), stable_tip_consensus_hash).map_err(net_error::DBError)? {
+                        Some(sn) => {
+                            if sn.block_height < highest_block_height {
+                                test_debug!("{:?}: neighbor {:?} is unstable, and has processed only up to stable burn block {} (we are targeting {})", &self.local_peer, nk, sn.block_height, target_block_height);
+
+                                highest_block_height = sn.block_height;
+                                num_blocks = if highest_block_height >= target_block_height { highest_block_height - target_block_height } else { 0 };
+                            }
+                        },
+                        None => {
+                            if *stable_tip_height <= target_block_height {
+                                // this peer has diverged from us
+                                test_debug!("{:?}: neighbor {:?}'s highest stable consensus hash {:?} does not match any of ours", &self.local_peer, nk, stable_tip_consensus_hash);
+                                return Ok(None);
+                            }
+                            else if *tip_height <= target_block_height {
+                                // this peer is unstable
+                                test_debug!("{:?}: neighbor {:?}'s highest stable consensus hash {:?} does not match our latest", &self.local_peer, nk, stable_tip_consensus_hash);
+                                return Ok(None);
+                            }
+                            // otherwise, this peer is simply ahead of us.
+                        }
+                    },
+                }
+            },
+            // never talked to this peer before
+            (_, _) => {}
+        }
+
+        if num_blocks == 0 {
+            test_debug!("{:?}: will not request BlocksInv from {:?}, since we are sync'ed up to its highest sortition block", &self.local_peer, nk);
+            return Ok(None);
+        }
+        assert!(num_blocks <= BLOCKS_INV_DATA_MAX_BITLEN as u64);
+
+        let mut tx = burndb.tx_begin().map_err(net_error::DBError)?;
```

Comment:
While purely a stylistic change (since the `?` operator will do the exact same call anyways), we should implement `From<SqliteError> for net::Error`, since it would eliminate ~150 calls of `map_err(net_error::DBError)`

Reply:
Sure, but should it be done in this PR?

Reply:
Nope, doesn't need to be in this PR